exec_syscall.c

```diff
@@ -0,0 +1,13 @@
+#include <common.h>
+#include <exec.h>
+
+int main() {
```

Comment:
Split it out to another PR, where you add `exec_launch` that can start arbitrary embedded process image based upon kernel parameter.